integration/benchmark/cmd/manual/main.go

```diff
@@ -138,17 +142,26 @@ func main() {
 			lg.Start()
 		}
 
-		// if the duration is 0, we run this case forever
+		var runDuration time.Duration
```

Comment:
to avoid the need to have magical constants you can initialize the channel to something that never closes, and re-assign it only if duration is positive:
```suggestion
		waitC := make(<-chan time.Time)
		if c.duration != 0 {
			waitC = time.After(c.duration)
		}
```

Reply:
I wasn't sure if it was an anti-pattern to make a channel that never closes. Thanks for the info, this is a cool technique.